addons/l10n_in_gstin_status/models/res_partner.py

```diff
@@ -34,6 +34,9 @@ def _onchange_l10n_in_gst_status(self):
 
     def action_l10n_in_verify_gstin_status(self):
         self.ensure_one()
+        self.check_access('write')
```

Comment:
I don't understand this line. Why are we checking that ?

Reply:
Here at the end of the method we're going to write the GSTIN status on the partner, if user has no access we will still do the IAP call then raise an AccessError, by doing this before the IAP call we prevent to unnecessarily consume IAP credits.